File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 4/3/2026, 2:02:34 PM
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        best, maxProfit = prices[0], 0

        for i in range(1, len(prices)):
            profit = prices[i] - best
            maxProfit = max(maxProfit, profit)
            best = min(best, prices[i])
        return maxProfit

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.cpp
// Last updated: 4/3/2026, 2:02:29 PM
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int best = prices[0], maxPro = 0;
        for(int i = 1; i < prices.size(); i++) {
            int curr = prices[i];
            maxPro = max(maxPro, curr-best);
            best = min(prices[i], best);
        }
        return maxPro;
    }
};